### ISSUE TITLE
Failed to add reference to 'libz3' while updating to Prose 2.0.0	

### ISSUE THREAD

**Issue by gustavoasoares** [Microsoft/prose#11]
It was not possible to update to update to Prose 2.0.0. When I try to do it using Nuget, I got the following error: 

Severity	Code	Description	Project	File	Line	Suppression State
Error		Failed to add reference to 'libz3'.			0	

Steps to reproduce: 
1. Open Nuget
2. Update Prose from 1.1.0 to 2.0.0

**Comment by simmdan** [comment]
First, I'm suspicious that the issue is related to the particular version of nuget you are using.  We may have a dependency on a newer version of nuget.  Can you tell me what version you are using?  If you are running nuget from the commandline you can find it from `nuget.exe --version`, and if you are running from within visual studio, then you should be able to get the version number of the nuget addin from tools/extensions.

Second, we discovered late yesterday that there's another problem with the 2.0.0 release, and we're working to get a 2.0.1 release out as soon as possible (maybe tonight).  I recommend that you wait to upgrade until that comes out.

**Comment by gustavoasoares** [comment]
Thank you, @simmdan! I am using the visual studio addin. The version is: 3.4.4.1321

I will wait for the version 2.0.1.

**Comment by simmdan** [comment]
Even when we have version 2.0.1 that probably won't address the issue in this thread.  So it would be good if you tried installing the latest nuget addin for vs: https://dist.nuget.org/visualstudio-2015-vsix/latest/NuGet.Tools.vsix (which is v3.5).  I'm not 100% positive, but I think that may address the issue.  If it does, please let me know, and we'll update future nuget packages to note in the metadata that we require that version.  If it doesn't, also please let me know so we can keep working to figure it out.

**Comment by simmdan** [comment]
Release 2.0.1 is now available.

**Comment by yangchunke** [comment]
I'm wondering if the issue has been resolved as I'm having the same issue while trying to install latest version on VS2015 Update 3 (NuGet PM 3.5.0)

PM> Install-Package Microsoft.ProgramSynthesis -Version 2.3.0

The installation failed with the error below.

Install-Package : Failed to add reference to 'libz3'.
At line:1 char:1
Install-Package Microsoft.ProgramSynthesis -Version 2.3.0
CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
FullyQualified

Thanks

**Comment by apskim** [comment]
@yangchunke This is a [known issue in VS 2015](https://microsoft.github.io/prose/release-notes/#release-230--20170515), but in VS 2017 installation should work without problems.

**Comment by simmdan** [comment]
Actually it appears that the issue occurs in VS 2017 as well, but we have determined that the problem is a combination of an issue with the Nuget UI in VS and our use of the native z3 library.  If you manually modify the nuget project configuration, then nuget restore and build will work fine.  This is easiest to do in VS 2017 where you can add a single PackageReference line to the csproj file--something like this:

```
<PackageReference Include="Microsoft.ProgramSynthesis" Version="2.3.0" />
```

If you are using VS 2015, then you can add a line to project.json (if you are using that) or packages.config.  With packages.config, though, you will need to also add lines for the nuget packages we depend on, and you will need to add DLL references in your csproj file.

We are working on the problem and hope to have a fix soon.  Sorry it has taken so long already.

**Comment by gustavoasoares** [comment]
Thanks, @simmdan! Worked for me!

**Comment by danpere** [comment]
We released [PROSE v3.0.0](https://www.nuget.org/packages/Microsoft.ProgramSynthesis/3.0.0) today, which fixes this issue. If you have further problems installing the NuGet package, please open a new issue.
